marshmallow/schema.py

```diff
@@ -99,6 +101,37 @@ def __new__(mcs, name, bases, attrs):
         for attr in mcs.FUNC_LISTS:
             attr_copy = copy.copy(getattr(klass, attr))
             setattr(klass, attr, attr_copy)
+
+        # Now add in the decorated processors. By doing this after constructing
+        # the class, we let standard inheritance do all the hard work.
+        mro = klass.mro()
+
+        klass.processors = defaultdict(list)
+        for attr_name in dir(klass):
```

Comment:
`dir` apparently returns an alphabetical list of attributes, meaning that decorated methods might not be applied in order expected by the developer. You might use `inspect` or store a counter on the decorators to get the right order.